practice/validator/site.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+/**
+ * Tests whether the string matches the regular expression
+ * @param  {string} testString The string to test
+ * @param  {reg exp} regExp    The regular expression.  Not a string.
+ * @return {boolean}           True if there is a match, false otherwise.
+ */
+function stringValidator(testString, regExp) {
+  return regExp.test(testString);
+}
+
+/**
+ * The the string entered on the form.
+ * @param  {jQuery selector} entryItem The selector for a text input.
+ * @return {string}                    The string entered on the form.
+ */
+function getFieldEntry(entryItem) {
+  return entryItem.val();
+}
+
+/**
+ * displayInvalid() highlights the input box and un-hides the warning message.
+ */
+function displayInvalid(entryItem) {
+  entryItem.addClass('invalid');
+  $(entryItem).next().addClass('visible-warning');
```

Comment:
This part may be a little obtuse.  There is a warning text right next to the box in the HTML, but is being hidden unless this class is applied.  I wanted to make sure the warning text was being allocated space by flex, otherwise I would have added/deleted the warning text entirely.